### DISCUSSION TITLE
When I connect to postgres rds database, I have got errors

### DISCUSSION THREAD

**Discussion by B1Z1** [strapi/strapi#6284] [Legacy Topics]
**Description**

When I try to connect to RDS db Amazon in production mode, like in instruction in documentation of strapi, I got errors. I have done all tasks from documentation, but it doesn't work.

**Steps**
1. Do all steps from https://strapi.io/documentation/3.0.0-beta.x/deployment/amazon-aws.html
2. Create ecosystem.config.js
3. Add environment variables to connect into remote rds db
4. Run "pm2 start ecosystem.config.js" 

**Expected**
Connect to database and run production build

0|strapi  | > backend@0.1.0 start /home/ubuntu/spokoj-backend
0|strapi  | > strapi start
0|strapi  | [2020-05-19T21:44:11.776Z] debug ⛔️ Server wasn't able to start properly.
0|strapi  | [2020-05-19T21:44:11.779Z] error TimeoutError: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
0|strapi  |     at Bluebird.try.then.catch (/home/ubuntu/spokoj-backend/node_modules/knex/lib/client.js:318:17)
0|strapi  |     at tryCatcher (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/util.js:16:23)
0|strapi  |     at /home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/catch_filter.js:17:41
0|strapi  |     at tryCatcher (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/util.js:16:23)
0|strapi  |     at Promise._settlePromiseFromHandler (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/promise.js:547:31)
0|strapi  |     at Promise._settlePromise (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/promise.js:604:18)
0|strapi  |     at Promise._settlePromise0 (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/promise.js:649:10)
0|strapi  |     at Promise._settlePromises (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/promise.js:725:18)
0|strapi  |     at _drainQueueStep (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/async.js:93:12)
0|strapi  |     at _drainQueue (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/async.js:86:9)
0|strapi  |     at Async._drainQueues (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/async.js:102:5)
0|strapi  |     at Immediate.Async.drainQueues [as _onImmediate] (/home/ubuntu/spokoj-backend/node_modules/bluebird/js/release/async.js:15:14)
0|strapi  |     at runCallback (timers.js:705:18)
0|strapi  |     at tryOnImmediate (timers.js:676:5)
0|strapi  |     at processImmediate (timers.js:658:5)
0|strapi  | npm
0|strapi  |
0|strapi  | ERR!
0|strapi  |
0|strapi  | code
0|strapi  |  ELIFECYCLE
0|strapi  | npm ERR! errno 1
0|strapi  | npm ERR! backend@0.1.0 start: `strapi start`
0|strapi  | npm ERR! Exit status 1
0|strapi  | npm ERR!
0|strapi  | npm ERR! Failed at the backend@0.1.0 start script.
0|strapi  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
0|strapi  | npm
0|strapi  |
0|strapi  | ERR!
0|strapi  |  A complete log of this run can be found in:
0|strapi  | npm ERR!     /home/ubuntu/.npm/_logs/2020-05-19T21_44_11_803Z-debug.log
PM2       | App [strapi:0] exited with code [1] via signal [SIGINT]
PM2       | App [strapi:0] starting in -fork mode-
PM2       | App [strapi:0] online

- Node.js version: v10.20.1
- NPM version: 6.14.5
- Strapi version: 3.0.0-beta.20.3
- Database: postgres 10.12
- Operating system: Ubuntu

![Annotation 2020-05-19 234538](https://user-images.githubusercontent.com/38375468/82381680-ea64cd00-9a2a-11ea-9417-161f13879f85.jpg)

Can you help me with the trouble, please! Thank you!

**Comment by derrickmehaffy** [comment]
```
Node.js version: latest
NPM version: latest
Strapi version: latest
```

Please properly fill out the template and format it to make it readable. The version lists at the bottom should be exact versions as "latest" tells us nothing.

> This is a templated message

Hello, please follow the issue template.

A proper issue submission let's us better understand the origin of your bug and therefore help you.

I will reopen your issue when we receive the issue following the template guidelines and properly fill out the template.

Please update the issue.

Thank you.

> A copy of the bug report template can be found below:

```
<!--
Hello 👋 Thank you for submitting an issue.

Before you start, please make sure your issue is understandable and reproducible.
To make your issue readable make sure you use valid Markdown syntax.

https://guides.github.com/features/mastering-markdown/
-->

**Describe the bug**
A clear and concise description of what the bug is.

**Steps to reproduce the behavior**
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Code snippets**
If applicable, add code samples to help explain your problem.

**System**
- Node.js version: <!-- Please ensure you are using the Node LTS version (v10/v12) -->
- NPM version:
- Strapi version: <!-- Please make sure you are on the latest version -->
- Database:
- Operating system:

**Additional context**
Add any other context about the problem here.
```

**Comment by B1Z1** [comment]
@derrickmehaffy open this issue please. I need solution

**Comment by derrickmehaffy** [comment]
@B1Z1 please see the pool config docs here: https://strapi.io/documentation/3.0.0-beta.x/concepts/configurations.html#database

It would appear your connection to the PG database is too slow and connections are timing out

**Comment by B1Z1** [comment]
@derrickmehaffy I increase pool max to 20, and timeoutTime to 10 seconds. But I have the same problem

**Comment by derrickmehaffy** [comment]
Try the following (which is what I use on my own production application):

```
      "options": {
        "acquireConnectionTimeout": 100000,
        "pool": {
          "min": 0,
          "max": 20,
          "createTimeoutMillis": 30000,
          "acquireTimeoutMillis": 600000,
          "idleTimeoutMillis": 30000
        }
      }
```

There is no promises that will work, but it would appear your connection to the database is quite slow. Or alternatively the database has far too few resources.

**Comment by B1Z1** [comment]
Nothing. I will try to recreate db with more resources

![Annotation 2020-05-23 114211](https://user-images.githubusercontent.com/38375468/82727391-83753b80-9cea-11ea-9f3a-d561e2bbfd45.jpg)

**Comment by derrickmehaffy** [comment]
As a side point (not directly related to this issue). If you are on AWS you should avoid running Strapi on the default `ubuntu` user as that is a special account that effectively acts as 'root'. You are bound to have multiple permission issues, among numerous other problems including multiple security ones by using that account.

My suggestion is you create a special service account to handle running Strapi.

**Comment by B1Z1** [comment]
I already use another account to work with strapi dashboard and database. I have created IAM user with perms to rds, ec2 and s3 bucket

**Comment by derrickmehaffy** [comment]
> I already use another account to work with strapi dashboard and database. I have created IAM user with perms to rds, ec2 and s3 bucket

Yeah I just noticed the path to which your npm is logging information was the `ubuntu` home:
![image](https://user-images.githubusercontent.com/8593673/82727588-408e7600-9ca0-11ea-9037-06359b2c2367.png)

**Comment by B1Z1** [comment]
At current time, I'm using for tests, but thank you :)

**Comment by B1Z1** [comment]
I was setup new RDS Database with 8GB RAM and 20GB SSD, but it doesn't work

**Comment by derrickmehaffy** [comment]
> I was setup new RDS Database with 8GB RAM and 20GB SSD, but it doesn't work

It's possible there is a problem connecting to the database, beyond that I would need to test it myself which I will try tomorrow if I get some time.

**Comment by B1Z1** [comment]
Any news @derrickmehaffy ?

**Comment by lauriejim** [comment]
Did you make sure the security group of your RDS allow connections from your instance IP?
